### PR TITLE
Add first NB material + images (WIP; do NOT merge)

### PR DESCRIPTION
**Pull request by hugobowne**

### CODE REVIEW

**Comment by hugobowne** [comment]
@dhavide 

request for feedback on this early first pass of forward propagation for single and multilayer perceptrons + gradient descent.

**I really just want a sanity check and any corrections wrt the code**, not the descriptions/words yet, which I'll work on more, and I'll also include some/more equations.

I'm also aware that I need to clean up the code, name variables a bit more cleverly.

Then I'll start to divide the material into instructional code + hands-on code for learners to do themselves.

I also plan to use more "real-world" data or to simulate some that will be for a real ML question.

In addition, I'll soon include a bunch of `scikit-learn` code for the intro. This will be straightforward.

**Comment by dhavide** [comment]
@hugobowne 

+ The narrative makes sense (code-wise) thus far.
+ I recommend focusing on a particular (say, binary) classification problem first, and building up to a multi-class classification. We can work in regression later if we have time, but getting the mechanics on a simple problem unambiguously clear makes most sense. Let's avoid introducing complexity unless necessary. When introducing activation functions, this is confusing ("Why ReLu vs. tanh vs. sigmoid?") but of we can provide a working example first, we can work our way up.
+ I'm not sure about mentioning representation learning. If we're not going to do it explicitly, then this extraneous detail is distracting more than helpful.
+ Backpropagation is extremely difficult to explain; how much of it we explain depends on prerequisites. If we're assuming essentially no prerequisite multivariate calculus knowledge, we should avoid the details completely (it will overwhelm without any real pedagogical benefit). If we assume some multivariate calculus (which is reasonable for the SciPy audience), we can make reference to the (multivariate) chain rule and *maybe* walk through an example of how it works. But, honestly, it's not clear what the benefit is of doing so... unless we construct an example to work through "the hard way" and show it again later in PyTorch "the easy way". That would actually have merit.
+ Gradient descent: that may be worth illustrating in code in a general context graphically for a simple 3D or 3D optimization problem. The challenge is implementing the gradient function explicitly (which is what automatic differentiation circumvents) but we can provide that. I likely have some examples kicking around from an undergraduate optimization course I taught years ago. Let me know if you want me to dig those up (we'd have to translate Matlab into Python/NumPy, but it shouldn't be hard).

**Comment by hugobowne** [comment]
@dhavide i've made significant changes and merged. lmk if there's anything I have **not** covered that's required for you to proceed with the 2nd half. Feel free to raise an issue in repo to do so.

onward!
